### DISCUSSION TITLE
How do I aggregate articles by tags?

### DISCUSSION THREAD

**Discussion by dongrentianyu** [web-infra-dev/rspress#492] [Q&A]
比如我有一些文章，是mdx文件，我能不能给它们加上tags字段，”精选“，然后在顶部写一个精选栏，然后下面的内容完全自动处理。所有具有”精选“的文章都会像正常文档一样展示在下面。

**Comment by sanyuan0704** [comment]
可以再详细描述下？

**Comment by dongrentianyu** [comment]
![image](https://github.com/web-infra-dev/rspress/assets/109711900/b750a184-fcac-4011-8f40-1ecd080bcfe6)

就是比如这两首诗歌，我觉得写得很好，想放在精选栏里。但我不想在精选栏里再重复写mdx，我想通过标签的形式，能够自动聚合到。下次我写了其他的诗歌，觉得可以放在精选里，我就只需要加上精选的标签就可以了。不需要再把mdx文件复制一遍。

**Reply by sanyuan0704** [reply]
直接通过 MDX 语法：

```tsx
// example.mdx

import A from '../tags/a.mdx'

<A />
```

**Reply by dongrentianyu** [reply]
但这样写，只能是把所有精选的文章放到了一个页面下。不能展示成多篇文章的形式。

![image](https://github.com/web-infra-dev/rspress/assets/109711900/f92f4e60-21ca-46c1-bf5a-550b2db3e2c8)

就侧边栏上只显示了一篇文章。

或者我要每一篇精选文章都写一个mdx文章的话，那就不如直接复制了。而且旁边的标题也是一个困难。

就还是更自动化一点才好。标签也方便展示不同类型的文章。

**Reply by dongrentianyu** [reply]
仔细看了文档，感觉我需要的应该是某种动态路由。就是完全通过标签来生成路由。可能需要自己写代码了来实现。但可惜的是不太会写代码。这个问题还是先搁置好了。
